src/server/api/routers/eventRoute.ts

```diff
@@ -0,0 +1,214 @@
+            })
+        )
+        .mutation(async (req) => {
+            const userCheck = await prisma.user.findUnique({
+                where: {
+                    email: req.input.email,
+                },
+            });
+            if (!userCheck) {
+                throw new TRPCError({
+                    code: 'NOT_FOUND',
+                    message: 'User does not exist',
+                  });
+            }
+            const newEvent = await prisma.event.create({
+                data: {
+                    occuringAt: new Date(), //"2020-01-01T00:00:00Z"
+                    ownerID: userCheck.id,
+                    address: req.input.address,
+                },
+            });
+            await prisma.participate.create({
+                data: {
+                    eventID: newEvent.id,
+                    userID: userCheck.id,
+                    timeSlots: {} = {},
+                },
+            });
+            return newEvent;
+        }),
+    /*
+        async function that get whole event list of an user.
+        First check the existance of user, then return the list of event objects. 
+    */
+    getEventList: protectedProcedure
+        .input(z.object({ email: z.string().email() }))
+        .query(async (req) => {
+            const userCheck = await prisma.user.findUnique({
+                where: {
+                    email: req.input.email,
+                },
+            });
+            if (!userCheck) {
+                throw new TRPCError({
+                    code: 'NOT_FOUND',
+                    message: 'User does not exist',
+                });
+            }
+            return await prisma.event.findMany({
+                where: { ownerID: userCheck.id },
+                select: { id: true, occuringAt: true},
+                orderBy: { occuringAt: 'asc' }
+            });
+        }),
+        
+    getEvent: protectedProcedure
+        .input(z.object({ 
+            eventId: z.string()
+        }))
+        .query(async (req) => {
+            const eventCheck = await prisma.event.findUnique({
+                where: {
+                    id: req.input.eventId,
+                },
+            });
+            if (!eventCheck) {
+                throw new TRPCError({
+                    code: 'NOT_FOUND',
+                    message: 'Event does not exist',
+                });
+            }
+            return eventCheck;
+        }),
+
+    /**
+     * Verify the status of event host and check the existance of event. 
+     * Then update event address.
+     */
+    updateEvent_addresss: protectedProcedure
+        .input(z.object({
+            host_email: z.string().email(),
+            address: z.string(),
+            eventId: z.string()
+        }))
+        .mutation(async (req) => {
+            const eventCheck = await prisma.event.findUnique({
+                where: {
+                    id: req.input.eventId,
+                },
+            });
```

Comment:
I believe for the check for eventOwner can somehow be done here in the top most query by giving it something like this:
```typescript
          ownerID: req.input.host_email,
```
source:https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries